src/librustdoc/html/render.rs

```diff
@@ -1292,7 +1375,7 @@ impl<'a> SourceCollector<'a> {
         layout::render(&mut w, &self.scx.layout,
                        &page, &(""), &Source(contents),
                        self.scx.css_file_extension.is_some(),
-                       &self.scx.themes)?;
+                       &self.scx.themes, &["source-files", "source-script"])?;
```

Comment:
See how these files are treated the same here? This is what i'm talking about in the other comment - one of these is the same for every doc build and is saved with the resource suffix, and the other is different for each crate and doesn't get the resource suffix. You either need a way to treat these differently in the `render` function, or you need to pull in the suffix right here and add it to `source-script`.